habitat/tasks/nav/nav_task.py

```diff
@@ -396,19 +421,27 @@ class SPL(Measure):
     https://arxiv.org/pdf/1807.06757.pdf
     """
 
-    def __init__(self, sim: Simulator, config: Config):
+    def __init__(
+        self,
+        *args: Any,
+        sim: Simulator,
+        task: EmbodiedTask,
```

Comment:
I think this will end up casing a circular reference i.e. the task holds a reference to the measure and the measure holds a reference to the task, and thus you can't every free the memory used by the task.  (I believe this is also a problem for the actions and the sensors)

There are a couple ways to deal with this:
1. The environment is the owner of actions/sensors/measures -- no circular references here.  Kinda a clunky design tho.
1. Use `weakref.proxy` when giving the measure -- gets rid of the circular here by giving the X a non-owning reference to the task.  This is kinda a weird solution tho.
1. Different API where there is a `task_state` that gets manipulate/populated by actions, then sensors, then measures.  It is passed in as an argument to their step/get/update functions.

The second one feels the most clean.  1 and 2 are both band-aids where the first makes the internals less clean and the second uses a kinda weird python API.

Reply:
That's valuable point. I expect generational garbage collector should take care of this as soon as there is no outside reference to the task. I will pass task in `reset` and `step` method as param to avoid this by design.

Reply:
Yeah, looking at this again, this does look like one of cases where the garbage collector can deal with this.